src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/mturk/MturkPlatform.java

```diff
@@ -0,0 +1,223 @@
+import com.amazonaws.mturk.requester.doc._2014_08_15.AssignmentStatus;
+import com.amazonaws.mturk.requester.doc._2014_08_15.BonusPayment;
+import edu.kit.ipd.crowdcontrol.objectservice.crowdworking.*;
+import edu.kit.ipd.crowdcontrol.objectservice.crowdworking.mturk.command.*;
+import edu.kit.ipd.crowdcontrol.objectservice.proto.Experiment;
+import edu.kit.ipd.crowdcontrol.objectservice.proto.Tag;
+
+import java.math.BigDecimal;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+
+/**
+ * Platform implementation for a mturk instance
+ * @author MarcelHollerbach
+ * @version 0.1
+ */
+public class MturkPlatform implements Platform,Payment,WorkerIdentification {
+    private final String name;
+    private final MTurkConnection connection;
+
+    /**
+     * A new mturk platform instance
+     * @param user user to login
+     * @param password password to use
+     * @param url instance to connect to
+     */
+    public MturkPlatform(String user, String password, String url, String name) {
+        connection = new MTurkConnection(user, password, url);
+        this.name = name;
+    }
+
+    @Override
+    public String getName() {
+        return "Mturk "+name;
+    }
+
+    @Override
+    public String getID() {
+        return "mturk"+name.toLowerCase();
+    }
+
+    @Override
+    public Boolean isCalibrationAllowed() {
+        return false;
+    }
+
+    @Override
+    public Optional<Payment> getPayment() {
+        return Optional.of(this);
+    }
+
+    @Override
+    public Optional<WorkerIdentification> getWorker() {
+        return Optional.of(this);
+    }
+
+
+    @Override
+    public CompletableFuture<String> publishTask(Experiment experiment) {
+        String tags = experiment.getTagsList().stream().map(Tag::getName).collect(Collectors.joining(","));
+        return new PublishHIT(connection,experiment.getTitle(),experiment.getDescription(),
+                experiment.getPaymentBase().getValue()/100d, //we are getting cents passed and have to pass dallers
+                60*60*2, //you have 2 hours to do the assignment
+                60*60*24*30, // the experiment is staying for 30 days
+                tags,
+                experiment.getNeededAnswers().getValue()*experiment.getRatingsPerAnswer().getValue(),
+                2592000, //this is a little problem we have to specify when autoapproval is kicking in this is happening after 2592000s
+                "");
+    }
+
+    @Override
+    public CompletableFuture<Boolean> unpublishTask(String id) {
+        return new UnpublishHIT(connection, id);
+    }
+
+    @Override
+    public CompletableFuture<String> updateTask(String id, Experiment experiment) {
+        return null;
+    }
+
+    @Override
+    public String identifyWorker(Map<String, String[]> param) throws UnidentifiedWorkerException {
+        return null;
```

Comment:
will put a fixme here cannot do anything here, since I dont have a worker-ui yet, where I have to wait for.